### DISCUSSION TITLE
Update time proposal

### DISCUSSION THREAD

**Discussion by Mister-Hope** [DreamOfIce/vuepress-plugin-i18n#7] [Ideas]
I don't think `calcUpdatedTime` is a good name, and it's confusing.

I have a proposal to change this with `updateTime: "git" | "file" | ((page: Page) => number | Date | undefined)`, with a default value of `(page)=> page.data.git?.updateTime`

Not all project are using git, the plugin could make a fallback for noob users to get updateTime from fileinfo if possible with the keyword `"file"`

Also with the built-in keyword `"git"`, the plugin should read ` page.data.git.updateTime`, and generate timestamp itself if the value does not exisits.

The function also gives the user ability to handle complicated cases (e.g: with git submodule and other situations where users want to define a update time themselves like frontmatter.updateTIme)

**Comment by Mister-Hope** [comment]
If you like the idea, I think you have the ability to done it yourself.

**Comment by DreamOfIce** [comment]
Sounds great!
But it's a little late today, I'll think about it tomorrow

**Comment by DreamOfIce** [comment]
Also, I think there should be an `auto` mode and use it as a default. Prefer to use git and use file info as fallback

**Reply by Mister-Hope** [reply]
I personally do not think file info could be a fallback, as when you are using git and checkout through branches and commits, the file edit time does not mean the time they are last edited. Also there will be cases where people move or rename files, if you do think there should be an auto mode, I strongly recommended setting a minimal time gap.

**Reply by DreamOfIce** [reply]
> as when you are using git and checkout through branches and commits, the file edit time does not mean the time they are last edited.

I think git mode should work in this case

**Comment by DreamOfIce** [comment]
completed in 7a677cdf94b31f2201861b4fe53ce8fe0ee770f7
